### PR TITLE
SRE-164: Use Workload Identity for `upload-docs` action

### PR DESCRIPTION
**Pull request by aleks-andr**
We want to enable Workload Identity for all Actions still using old SA keys. `upload-docs` appears to be the only action in this repository not yet supporting WI; this PR adds that support. This will most likely require a major version bump.

### CODE REVIEW

**Review by mihaitabamburic — APPROVED** [review]

**Comment by aleks-andr** [comment]
@mihaitabamburic I didn't realize at first but this appears to be a duplicate of #40 🤦  Want me to merge this or should we instead keep the other implementation?

**Comment by pkmetski** [comment]
> @mihaitabamburic I didn't realize at first but this appears to be a duplicate of #40 🤦 Want me to merge this or should we instead keep the other implementation?

Maybe @emmanuelstroem can say? The two implementations are rather different and I am not sure why.

**Comment by aleks-andr** [comment]
Looking at the two, they are functionally pretty much identical. Emmanuel's version makes consumers provide more parameters, but mine keeps the "API" almost the same. I'm not sure which one is better here: the added flexibility or just a "one-click" solution to upload docs. If you prefer the added flexibility, just let me know and I'll close this PR and move the README changes to the other one.

**Comment by pkmetski** [comment]
@aleks-andr what is the state of this pr?

**Comment by aleks-andr** [comment]
> @aleks-andr what is the state of this pr?

Should be ready, just awaiting a review from you guys 😄

File: upload-docs/action.yml [hunk 1]
```diff
@@ -2,20 +2,20 @@ name: Upload docs
...
 
 inputs:
-  docs_bucket_sa:
-    description: "Documentation bucket service account"
+  workload_identity_pool_provider:
+    description: "ID of workload identity federation pool provider"
     required: true
+  service_account_email:
+    description: "Google Service Account Email with GCS bucket permissions"
+    required: true
```
Comment (pkmetski) [inline upload-docs/action.yml:10]: This will require a major version bump

**Review by pkmetski — APPROVED** [review]
LGTM

**Pull request merged**